internal/p2p/peermanager.go

```diff
@@ -513,12 +513,21 @@ func (m *PeerManager) HasDialedMaxPeers() bool {
 // returned peer.
 func (m *PeerManager) DialNext(ctx context.Context) (NodeAddress, error) {
 	for {
-		address, err := m.TryDialNext()
-		if err != nil || (address != NodeAddress{}) {
-			return address, err
+		if address := m.TryDialNext(); (address != NodeAddress{}) {
+			return address, nil
 		}
+
+		if ctx.Err() != nil {
+			return NodeAddress{}, ctx.Err()
+		}
+
+		if !m.HasDialedMaxPeers() {
+			continue
+		}
+
 		select {
 		case <-m.dialWaker.Sleep():
+			continue
```

Comment:
Fine, but this is what it does anyway.

Reply:
yeah, I was just trying to make the paths through these functions a bit clearer.